### PR TITLE
Improve TextInput component in preparation for renaming columns

### PR DESCRIPTION
**Pull request by seancolsen**
## Notes

- These changes lay some of the groundwork needed for #497
- These changes were originally performed in 3d8352225e789a02448654975c63b39d15a84291 as part of #870. They're now in a new commit because the first one accumulated too many merge conflicts easily be merged.
- This commit is also now in its own PR so that we can merge this in without resolving the [complex discussions](https://github.com/centerofci/mathesar/pull/870#issuecomment-995174178) necessary to continue with implementing the column renaming.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: mathesar_ui/src/component-library/text-input/TextInput.svelte [hunk 1]
```diff
@@ -18,22 +18,30 @@
...
 
   // Underlying DOM element for direct access
-  export let element: HTMLElement = null;
+  export let element: HTMLInputElement | undefined = undefined;
 
   // Id for the input
   export let id: string = undefined;
 
-  function handleKeypress(e: KeyboardEvent) {
+  export let hasValidationErrors = false;
```
Comment (pavish) [inline mathesar_ui/src/component-library/text-input/TextInput.svelte:26]: We should handle this for all input components, but we can take it up in separate PRs when needed.

Comment (seancolsen) [reply mathesar_ui/src/component-library/text-input/TextInput.svelte:26]: I considered that too. Certainly for components that _compose_ TextInput, like NumberInput. But I'm not sure if we need it for any other ones. The only one to _maybe_ have it for is TextArea, but I can't really think of a realistic use case there.

Comment (seancolsen) [reply mathesar_ui/src/component-library/text-input/TextInput.svelte:26]: Oh Oops, NumberInput doesn't compose TextInput. I was thinking of previous work I did. We might consider the usefulness of doing so.

Comment (seancolsen) [reply mathesar_ui/src/component-library/text-input/TextInput.svelte:26]: Ah! Shoot -- sorry -- I was moving too fast. I looked at the diff and thought you were talking about `handleKeypress`.

For `hasValidationErrors`, yes, I agree this should be applicable to TextArea for sure. For other inputs, like Checkbox, Radio, and Select I think it's probably not relevant.

**Review by pavish — APPROVED** [review]
@seancolsen Looks good to me!

**Pull request merged**